pkg/verify/tlog_test.go

```diff
@@ -221,3 +228,68 @@ func TestMaxAllowedTlogEntries(t *testing.T) {
 	_, err = verify.VerifyArtifactTransparencyLog(&tooManyTlogEntriesEntity{entity}, virtualSigstore, 1, true, false)
 	assert.ErrorContains(t, err, "too many tlog entries") // too many tlog entries should fail to verify
 }
+
+type mockEntriesClient struct {
+	Entries []*models.LogEntry
+}
+
+func (m *mockEntriesClient) CreateLogEntry(_ *entries.CreateLogEntryParams, _ ...entries.ClientOption) (*entries.CreateLogEntryCreated, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (m *mockEntriesClient) GetLogEntryByIndex(params *entries.GetLogEntryByIndexParams, _ ...entries.ClientOption) (*entries.GetLogEntryByIndexOK, error) {
+	resp := &entries.GetLogEntryByIndexOK{}
+	if len(m.Entries) != 0 {
+		for _, e := range m.Entries {
+			for _, i := range *e {
+				if *i.LogIndex == params.LogIndex {
+					resp.Payload = *e
+				}
+			}
+		}
+
+		if resp.Payload == nil {
+			resp.Payload = *m.Entries[0]
+		}
+	}
+	return resp, nil
+}
+
+func (m *mockEntriesClient) GetLogEntryByUUID(_ *entries.GetLogEntryByUUIDParams, _ ...entries.ClientOption) (*entries.GetLogEntryByUUIDOK, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (m *mockEntriesClient) SearchLogQuery(_ *entries.SearchLogQueryParams, _ ...entries.ClientOption) (*entries.SearchLogQueryOK, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (m *mockEntriesClient) SetTransport(_ runtime.ClientTransport) {}
+
+func TestTlogVerification(t *testing.T) {
```

Comment:
Could this be renamed to make it clear it's testing online verification:
```suggestion
func TestOnlineVerification(t *testing.T) {
```